src/pad_action_entry.rs

```diff
@@ -4,50 +4,53 @@
 
 use glib::translate::*;
 use gtk_sys;
+use std::ffi::CStr;
 use PadActionType;
 
-#[repr(C)]
-pub struct PadActionEntry(gtk_sys::GtkPadActionEntry);
+#[derive(Debug, Clone)]
+pub struct PadActionEntry {
+    pub(crate) type_: PadActionType,
+    pub(crate) index: i32,
+    pub(crate) mode: i32,
+    pub(crate) label: String,
+    pub(crate) action_name: String,
+}
 
 impl PadActionEntry {
-    pub fn new(type_: PadActionType, index: i32, mode: i32, label: &str,
-               action_name: &str) -> PadActionEntry {
+    pub fn new(
+        type_: PadActionType,
+        index: i32,
+        mode: i32,
+        label: &str,
+        action_name: &str,
+    ) -> PadActionEntry {
         assert_initialized_main_thread!();
-        PadActionEntry(gtk_sys::GtkPadActionEntry {
-            type_: type_.to_glib(),
+        PadActionEntry {
+            type_,
             index,
             mode,
-            label: label.to_glib_none().0,
-            action_name: action_name.to_glib_none().0,
-        })
+            label: label.to_owned(),
+            action_name: action_name.to_owned(),
+        }
     }
 
     pub fn get_type(&self) -> PadActionType {
-        from_glib(self.0.type_)
+        self.type_
     }
 
     pub fn get_index(&self) -> i32 {
-        self.0.index
+        self.index
     }
 
     pub fn get_mode(&self) -> i32 {
-        self.0.mode
-    }
-
-    pub fn get_label(&self) -> Option<String> {
-        unsafe { from_glib_none(self.0.label) }
+        self.mode
     }
 
-    pub fn get_action_name(&self) -> Option<String> {
-        unsafe { from_glib_none(self.0.label) }
+    pub fn get_label(&self) -> Option<&str> {
+        Some(&self.label)
     }
-}
-
-#[doc(hidden)]
-impl ToGlib for PadActionEntry {
-    type GlibType = gtk_sys::GtkPadActionEntry;
 
-    fn to_glib(&self) -> gtk_sys::GtkPadActionEntry {
-        self.0
+    pub fn get_action_name(&self) -> Option<&str> {
+        Some(&self.action_name)
```

Comment:
Even more so, these two string getters don't really need to return an `Option`. They can't be `None` ever